pkg/controller/controller.go

```diff
@@ -596,6 +604,14 @@ func (p *csiProvisioner) prepareProvision(ctx context.Context, claim *v1.Persist
 		}
 	}
 
+	vacName := claim.Spec.VolumeAttributesClassName
+	// TODO: For reviewers - Should we reject volumes that have a VAC specified if the VAC feature gate is disabled?
```

Comment:
The csi spec states that "A Controller plugin MUST implement this RPC call if the plugin has the MODIFY_VOLUME controller capability. " 

To consider backward compatibility I do not think we should reject the call. Maybe if there is a way to send a warning to users indicate that the VAC is not applied because of feature gate etc.

Reply:
Why would it not be backwards compatible? A user would have to opt-in to specifying the VAC in their PVC.

Reply:
What if they downgrade their cluster and turn off the vac feature but the VAC is applied?

Reply:
Kubernetes doesn't support downgrades, although I'm not 100% sure what would happen if you disabled the API Server feature gate on a cluster with VACs already in it

Reply:
Kubernetes supports downgrades. 

However, even with the help of the folks on #sig-architecture the best link that I could find was only the underlying design for API storage: https://kubernetes.io/docs/reference/using-api/deprecation-policy/

Reply:
Maybe "doesn't support" is too strong of wording, but Kubernetes does not document any support for downgrading and it is widely considered unsafe: https://github.com/kubernetes/website/issues/12327

That said, I'm not sure it really matters for the purpose of this question: My question is basically, if the volume does specify a `VolumeAttributeClassName` (i.e. the feature flag is enabled at the API level and the user is using it) but the local `external-provisioner` feature flag is disabled, what should we do?

Should we either (1) provision the volume as normal - ignoring the fact it has a VAC or (2) refuse to provision the volume and demand the user turns on the VAC feature flag or remove the VAC from the volume?

Reply:
Much better than my version, I stole it wholesale for the next revision